File: client/Assets/Scripts/Message/info/chenliang/moba/message/BattleToPlayerDispatcher.cs
namespace info.chenliang.moba.message{
using System;
using System.IO;
using MiscUtil;
using MiscUtil.IO;
using MiscUtil.Conversion;
using Talky;
public class BattleToPlayerDispatcher {
public static void Dispatch(MemoryEndianBinaryReader reader, BattleToPlayerProxy proxy) {
int fid = reader.ReadByte();
switch(fid)
{
case 0:
{
battlePing(reader, proxy);
}
break;
case 1:
{
battlePong(reader, proxy);
}
break;
case 2:
{
onJoinArena(reader, proxy);
}
break;
case 3:
{
sync(reader, proxy);
}
break;
case 4:
{
playerSpawned(reader, proxy);
}
break;
}
}
protected static void battlePing(MemoryEndianBinaryReader reader, BattleToPlayerProxy proxy){
proxy.battlePing();
}
protected static void battlePong(MemoryEndianBinaryReader reader, BattleToPlayerProxy proxy){
proxy.battlePong();
}
protected static void onJoinArena(MemoryEndianBinaryReader reader, BattleToPlayerProxy proxy){
Int32 arenaId;
Sync sync;
Int32 entityId;
arenaId = reader.ReadInt32();
sync = new Sync();
sync.Deserialize(reader);
entityId = reader.ReadInt32();
proxy.onJoinArena(arenaId,sync,entityId);
}
protected static void sync(MemoryEndianBinaryReader reader, BattleToPlayerProxy proxy){
Sync sync;
sync = new Sync();
sync.Deserialize(reader);
proxy.sync(sync);
}
protected static void playerSpawned(MemoryEndianBinaryReader reader, BattleToPlayerProxy proxy){
Int32 entityId;
SyncItem item;
entityId = reader.ReadInt32();
item = new SyncItem();
item.Deserialize(reader);
proxy.playerSpawned(entityId,item);
}
}
}

File: server/battle/src/main/java/info/chenliang/moba/message/BattleToPlayerDispatcher.java
package info.chenliang.moba.message;
import java.io.*;
import info.chenliang.talky.*;
public class BattleToPlayerDispatcher {
public static void dispatch(DataInputStream dis, BattleToPlayerProxy proxy) throws Exception{
int fid = (int)(dis.read()&0xFFFF);
switch(fid)
{
case 0:
{
battlePing(dis, proxy);
}
break;
case 1:
{
battlePong(dis, proxy);
}
break;
case 2:
{
onJoinArena(dis, proxy);
}
break;
case 3:
{
sync(dis, proxy);
}
break;
case 4:
{
playerSpawned(dis, proxy);
}
break;
}
}
protected static void battlePing(DataInputStream dis, BattleToPlayerProxy proxy) throws Exception{
proxy.battlePing();
}
protected static void battlePong(DataInputStream dis, BattleToPlayerProxy proxy) throws Exception{
proxy.battlePong();
}
protected static void onJoinArena(DataInputStream dis, BattleToPlayerProxy proxy) throws Exception{
int arenaId;
Sync sync;
int entityId;
arenaId = dis.readInt();
sync = new Sync();
sync.deserialize(dis);
entityId = dis.readInt();
proxy.onJoinArena(arenaId,sync,entityId);
}
protected static void sync(DataInputStream dis, BattleToPlayerProxy proxy) throws Exception{
Sync sync;
sync = new Sync();
sync.deserialize(dis);
proxy.sync(sync);
}
protected static void playerSpawned(DataInputStream dis, BattleToPlayerProxy proxy) throws Exception{
int entityId;
SyncItem item;
entityId = dis.readInt();
item = new SyncItem();
item.deserialize(dis);
proxy.playerSpawned(entityId,item);
}
}
